### PR TITLE
Declare a fuchsia_asset_builder sdk_atom.

### PR DESCRIPTION
**Pull request by alainv**

### CODE REVIEW

File: packages/flutter_tools/BUILD.gn [hunk 1]
```diff
@@ -75,3 +75,26 @@ dart_tool("fuchsia_tester") {
     "$flutter_root/shell",
   ]
 }
+
+sdk_atom("fuchsia_asset_builder_sdk") {
```
Comment (pylaligand) [inline packages/flutter_tools/BUILD.gn]: You're missing an import for this template.

File: packages/flutter_tools/BUILD.gn [hunk 2]
```diff
@@ -75,3 +75,26 @@ dart_tool("fuchsia_tester") {
...
+  fuchsia_asset_builder_path = "$dart_target_gen_dir/fuchsia_asset_builder.snapshot"
+
+  files = [
+    {
+      source = "$fuchsia_asset_builder_path"
+      dest = "dart_prebuilts/fuchsia_asset_builder.snapshot"
+    },
+  ]
+
+  non_sdk_deps = dart_sdk_deps + [
```
Comment (pylaligand) [inline packages/flutter_tools/BUILD.gn]: Where is dart_sdk_deps defined?

**Review by pylaligand — CHANGES_REQUESTED** [review]

File: packages/flutter_tools/BUILD.gn (hunk 1 above)
Comment (alainv) [reply packages/flutter_tools/BUILD.gn]: Done, not sure how it built in the first place.

File: packages/flutter_tools/BUILD.gn (hunk 2 above)
Comment (alainv) [reply packages/flutter_tools/BUILD.gn]: Done.

**Comment by Hixie** [comment]
cc @alainv @pylaligand are you still working on this PR?

**Comment by Hixie** [comment]
ping @pylaligand PTAL

File: packages/flutter_tools/BUILD.gn [hunk 3]
```diff
@@ -75,3 +77,26 @@ dart_tool("fuchsia_tester") {
     "$flutter_root/shell",
   ]
 }
+
+sdk_atom("fuchsia_asset_builder_sdk") {
+  domain = "dart"
+  name = "fuchsia_asset_builder"
+  category = "partner"
```
Comment (pylaligand) [inline packages/flutter_tools/BUILD.gn]: You're going to add an id:
```
id = "sdk://tools/dart/fuchsia_asset_builder"
```

File: packages/flutter_tools/BUILD.gn [hunk 4]
```diff
@@ -75,3 +77,26 @@ dart_tool("fuchsia_tester") {
     "$flutter_root/shell",
   ]
 }
+
+sdk_atom("fuchsia_asset_builder_sdk") {
+  domain = "dart"
```
Comment (pylaligand) [inline packages/flutter_tools/BUILD.gn]: It would be easier to put this in the `exe` domain.

File: packages/flutter_tools/BUILD.gn (hunk 3 above)
Comment (alainv) [reply packages/flutter_tools/BUILD.gn]: Done.

File: packages/flutter_tools/BUILD.gn (hunk 4 above)
Comment (alainv) [reply packages/flutter_tools/BUILD.gn]: Done.

**Review by pylaligand — APPROVED** [review]

**Comment by alainv** [comment]
Can someone help merge this? Thanks!

**Comment by alainv** [comment]
Taking another approach.

**Pull request closed without merging**